### PR TITLE
[FIX] Bad token prevention

### PR DESCRIPTION
**Pull request by rksh**
Toast and Dialog presentation apparently causes occasional BadTokenExceptions - this should prevent app termination out from under display of modals.

### CODE REVIEW

**Review by bobzilladev — APPROVED** [review]
You would think they would have this check internally. Could we move the check to a helper method instead, passing in the interface that has isFinishing? Might be a little cleaner, remove the 'if' everywhere. Otherwise +1.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MappingFragment.java [hunk 1]
```diff
@@ -362,13 +363,10 @@ public Tile getTile(int x, int y, int zoom) {
 
                             try {
                                 final byte[] data = downloadData(new URL(s), userAgent, authToken);
-                                if (data != null) {
-                                    return new Tile(providerTileRes, providerTileRes, data);
-                                }
+                                return new Tile(providerTileRes, providerTileRes, data);
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MappingFragment.java:366]: data can actually never be null?

Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MappingFragment.java:366]: according to the linter!

Comment (bobzilladev) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MappingFragment.java:366]: obey

**Review by bobzilladev — APPROVED** [review]

**Pull request merged**